tests/notifications/test-class-yoast-notification-center.php

```diff
@@ -0,0 +1,451 @@
+
+		$notification = new Yoast_Notification( 'notification' );
+
+		$condition = $this
+			->getMockBuilder( 'Yoast_Notification_Condition' )
+			->setMethods( array( 'is_met', 'get_notification' ) )
+			->getMock();
+
+		$condition
+			->expects( $this->any() )
+			->method( 'is_met' )
+			->will( $this->returnValue( true ) );
+
+		$condition
+			->expects( $this->any() )
+			->method( 'get_notification' )
+			->will( $this->returnValue( $notification ) );
+
+		$subject = Yoast_Notification_Center::get();
+		$subject->add_notification_condition( $condition );
+
+		$this->assertEquals( array( $condition ), $subject->get_notification_conditions() );
+	}
+
+
+	/**
+	 * Registering a condition
+	 */
+	public function test_register_condition_twice() {
+
+		$notification = new Yoast_Notification( 'notification' );
+
+		$condition = $this
+			->getMockBuilder( 'Yoast_Notification_Condition' )
+			->setMethods( array( 'is_met', 'get_notification' ) )
+			->getMock();
+
+		$condition
+			->expects( $this->any() )
+			->method( 'is_met' )
+			->will( $this->returnValue( true ) );
+
+		$condition
+			->expects( $this->any() )
+			->method( 'get_notification' )
+			->will( $this->returnValue( $notification ) );
+
+		$subject = Yoast_Notification_Center::get();
+		$subject->add_notification_condition( $condition );
+		$subject->add_notification_condition( $condition );
+
+		$this->assertEquals( array( $condition ), $subject->get_notification_conditions() );
+	}
+
+
+	/**
+	 * Clear notification after setting
+	 */
+	public function test_clear_notifications() {
+		$notification = new Yoast_Notification( 'notification' );
+
+		$condition = $this
+			->getMockBuilder( 'Yoast_Notification_Condition' )
+			->setMethods( array( 'is_met', 'get_notification' ) )
+			->getMock();
+
+		$condition
+			->expects( $this->any() )
+			->method( 'is_met' )
+			->will( $this->returnValue( true ) );
+
+		$condition
+			->expects( $this->any() )
+			->method( 'get_notification' )
+			->will( $this->returnValue( $notification ) );
+
+		$subject = Yoast_Notification_Center::get();
+		$subject->add_notification_condition( $condition );
+
+		$subject->deactivate_hook();
+
+		$this->assertEquals( array(), $subject->get_notification_conditions() );
+	}
+
+
+	/**
+	 * Add notification
+	 */
+	public function test_add_notification() {
+		$notification = new Yoast_Notification( 'notification' );
+
+		$subject = Yoast_Notification_Center::get();
+		$subject->add_notification( $notification );
+
+		$this->assertEquals( array( $notification ), $subject->get_notifications() );
+	}
+
+	/**
+	 * Add wrong notification
+	 */
+	public function test_add_notification_twice() {
+		$notification = new Yoast_Notification( 'notification' );
+
+		$subject = Yoast_Notification_Center::get();
+		$subject->add_notification( $notification );
+		$subject->add_notification( $notification );
+
+		$notifications = $subject->get_notifications();
+
+		$this->assertEquals( 2, count( $notifications ) );
```

Comment:
You can add a stricter check: `$this->assertEquals( array( $notification, $notification ), $notifications );`